src/Marten/StoreOptions.cs

```diff
@@ -95,6 +95,11 @@ public string DatabaseSchemaName
         public int UpdateBatchSize { get; set; } = 500;
 
         /// <summary>
+        /// Configures the store to use char buffer pooling, greatly reducing allocations for serializing documents and events.
+        /// </summary>
+        public bool UseCharBufferPooling { get; set; } = false;
```

Comment:
In majority of cases I prefer to use `opt-in`,  then `opt-out` in a next version of a product. I'm not changing the behavior now as it's easier to test with no caching and focus on a few tests with caching. I'll add this as the point to this PR though.